### PR TITLE
{chem}[foss/2021b] Amber v22.0 w/ Python 3.9.6 + CUDA 11.4.1

### PR DESCRIPTION
**Pull request by bedroge**
(created using `eb --new-pr`)

This needs the easyblock from ~~https://github.com/easybuilders/easybuild-easyblocks/pull/2720~~ and the patch from ~~https://github.com/fwautele/easybuild-easyconfigs/pull/1~~ in order to properly detect/use FlexiBLAS.

### CODE REVIEW

**Comment by bedroge** [comment]
I copied `runtest = False` from the Amber 20 `fosscuda` easyconfigs. Just tried it with tests enabled, and three of them are failing for me:

```
166 file comparisons passed
3 file comparisons failed (1 of which can be ignored)
0 tests experienced errors

possible FAILURE: (ignored)  check mdout.mcwat.dif
/dev/shm/f115261/Amber/22.0/foss-2021b-AmberTools-22.3-CUDA-11.4.1/test/mcwat
104c104
<  Water move success.  Probability of movement:   0.56122010597643790
---
>  Water move success.  Probability of movement:   0.56122010597666983
106c106
<  Accepted probability:    6.5925926193774728E-2
---
>  Accepted probability:    6.5925926193739881E-2
---------------------------------------
possible FAILURE:  check mdout.dif
/dev/shm/f115261/Amber/22.0/foss-2021b-AmberTools-22.3-CUDA-11.4.1/test/kmmd/kmmd_pmemd
919c919
<  EAMBER (non-restraint)  =       124.42
---
>  EAMBER (non-restraint)  =       124.41
---------------------------------------
possible FAILURE:  check mdout.dif
/dev/shm/f115261/Amber/22.0/foss-2021b-AmberTools-22.3-CUDA-11.4.1/test/kmmd/kmmd_pmemd_gb
803c803
<  EELEC  =        42.26  EGB     =        40.14  RESTRAINT  =         1.20
---
>  EELEC  =        42.26  EGB     =        40.14  RESTRAINT  =         1.19
---------------------------------------
```

So, for now, I'll stick to `runtest = False`.

**Comment by akesandgren** [comment]
@bedroge See #16166 for my non-cuda version with all my patches listed.

**Comment by bedroge** [comment]
> @bedroge See #16166 for my non-cuda version with all my patches listed.

Thanks! I've synced this PR with yours and applied the same patches.

File: easybuild/easyconfigs/a/Amber/Amber-22.0-foss-2021b-AmberTools-22.3-CUDA-11.4.1.eb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    ('Boost', '1.77.0'),
+    ('libreadline', '8.1'),
+    ('matplotlib', '3.4.3'),
+    ('netCDF', '4.8.1'),
+    ('netCDF-Fortran', '4.5.3'),
+    ('PnetCDF', '1.12.3'),
+    ('Tkinter', '3.9.6'),
+    ('X11', '20210802'),
+    ('CUDA', '11.4.1', '', SYSTEM),
+    ('NCCL', '2.10.3', '-CUDA-%(cudaver)s'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/a/Amber/Amber-22.0-foss-2021b-AmberTools-22.3-CUDA-11.4.1.eb:92]: Should have checked this earlier, we also need UCX-CUDA/1.11.2-CUDA-11.4.1

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/a/Amber/Amber-22.0-foss-2021b-AmberTools-22.3-CUDA-11.4.1.eb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    'a86eee60bd65c16a849469e303cb99dfc207cbadd2ae9e70b9ff580ced785475',
+    '3fc08a2cf61f3eefd094554e60c88b14c9f5071f94e0a0a15c6b55fe321bfc78',  # AmberTools-22_fix_cuda_version_check.patch
+    'b5855866c63c2ee860a5e6ad6f618ac4dab7072f5caae95769d4b2457d616a8e',  # Amber-22_reduce_precision_of_kmmd_test.patch
+    # Amber-22_remove_undeclared_redundant_variable.patch
+    'b94900c2178dd6dbf2824b17074980a3e5e6e71b38c0b2b30e1f147e4e1ac8cb',
+]
+
+builddependencies = [
+    ('Bison', '3.7.6'),
+    ('CMake', '3.21.1'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/a/Amber/Amber-22.0-foss-2021b-AmberTools-22.3-CUDA-11.4.1.eb:72]: And while you're at it, swap these lines around to make it identical to the non-cuda version

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/a/Amber/Amber-22.0-foss-2021b-AmberTools-22.3-CUDA-11.4.1.eb (hunk 2 above)
Comment (akesandgren) [reply easybuild/easyconfigs/a/Amber/Amber-22.0-foss-2021b-AmberTools-22.3-CUDA-11.4.1.eb:72]: @bedroge ping

File: easybuild/easyconfigs/a/Amber/Amber-22.0-foss-2021b-AmberTools-22.3-CUDA-11.4.1.eb (hunk 1 above)
Comment (bedroge) [reply easybuild/easyconfigs/a/Amber/Amber-22.0-foss-2021b-AmberTools-22.3-CUDA-11.4.1.eb:92]: Done in https://github.com/easybuilders/easybuild-easyconfigs/pull/16123/commits/5a9bc1e2fdaf5ed66570a6b4805e0598ab4f3f91

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/1bdb63ba14a4733305cff7d6c4784c90 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/e1fcb74237871c02144f234f60e10140 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 18 out of 19 (1 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 470.161.03, Python 3.8.10
See https://gist.github.com/7cfde7f5447b4df71c412818f40e6d84 for a full test report.

**Comment by akesandgren** [comment]
I think we need to turn off testing in this easyconfig after all. Too much is going wrong for some reason.

**Comment by bedroge** [comment]
> I think we need to turn off testing in this easyconfig after all. Too much is going wrong for some reason.

I also had to disable them when installing this locally (and haven't received any complaints from the user who requested it afterwards), so I've just changed this in the easyconfig.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 20.04.5 LTS (Focal Fossa), x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 470.161.03, Python 3.8.10
See https://gist.github.com/c79cba87527a639b2795a6e49183c09d for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 470.161.03, Python 3.8.10
See https://gist.github.com/1b67b0c0b102fdf5d1ac6373c665e348 for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @bedroge!

**Pull request merged**